index.d.ts

```diff
@@ -81,7 +81,8 @@ export declare interface Tracer extends opentracing.Tracer {
    * If the `orphanable` option is set to false, the function will not be traced
    * unless there is already an active span or `childOf` option.
    */
-  trace<T> (name: string, fn: (span?: Span, fn?: (error?: Error) => any) => T): T;
+  trace<T> (name: string, fn: (span: Span) => T): T;
+  trace<T> (name: string, fn: (span: Span, done: (error?: Error) => string) => T): T;
```

Comment:
Interestingly enough, `docs/test.ts` already covers it [here](https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts#L321-L323).. and typescript has errors in `master`

![image](https://user-images.githubusercontent.com/877396/212760480-5bef0243-1ef9-49ff-8822-81d35411a3d8.png)

(whereas typescript doesn't complain on my `patch-1` branch)

Reply:
im using VSCode with a bunch of extensions.. but i think the only relevant one here is [TypeScript Hero](https://marketplace.visualstudio.com/items?itemName=rbbit.typescript-hero) -- can't see anything else that would contribute

Reply:
fwiw `docs/test.ts` has lots of other typescript errors -- at least according to my environment 
![image](https://user-images.githubusercontent.com/877396/212761688-c0189e30-3b6b-44c4-b117-34e55effa351.png)

Reply:
I see these errors as well in VS Code but not when running `yarn type:test` 🤔 Do you know what we're doing wrong that these aren't captured in the test? Ideally it would be fixed so that we don't end up with regressions for these cases.

Reply:
Probably vscode isn't using the same base tsconfig or typescript version.

Reply:
I can confirm that if I switch VSCode's TS to the version defined in `docs/package.json` it still shows errors (in VSCode)

- open a new VSCode window from `dd-trace-js/docs`
- run `yarn` from that directory (ie. the VSCode terminal)
- open `test.ts` - warnings still appear

Reply:
> Do you know what we're doing wrong that these aren't captured in the test?

Unfortunately I do not know